controllers/statusreport/statusreport_adapter.go

```diff
@@ -322,3 +325,17 @@ func (a *Adapter) findUntriggeredIntegrationTestFromStatus(integrationTestScenar
 	}
 	return ""
 }
+
+// RequeueIfYoungerThanThreshold checks if the adapter' snapshot is younger than the threshold defined
+// in the function.  If it is, the function returns an operation result instructing the reconciler
+// to requeue the object and the error message passed to the function.  If not, the function returns
+// an operation result instructing the reconciler NOT to requeue the object.
+func (a *Adapter) RequeueIfYoungerThanThreshold(retErr error) (controller.OperationResult, error) {
```

Comment:
Non-blocking suggestion: this function could be useful to other controllers too, so it might not be bad to move it to a package outside this adapter so it's more easily available elsewhere.

Reply:
+1 on moving this function to a common location so it can be used both by statusreport, snapshot ([as here](https://github.com/redhat-appstudio/integration-service/blob/037767a3f1f57fb6a1a1108dc01795b4e48bbabc/controllers/snapshot/snapshot_adapter.go#L942)), and other controllers.

Reply:
Also it can be generic for any object, not just Snapshot

Reply:
updated to be generic in most recent patch